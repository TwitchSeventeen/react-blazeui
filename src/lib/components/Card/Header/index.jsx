import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function Header(props) {
  const {
    children, className, dataName, ...other
  } = props;
  const classes = classNames('c-card__header', className);
  return (
    <header
      {...other}
      className={classes}
      data-name={dataName}
    >
      { children }
    </header>
  );
}

Header.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
  /**
   * Additional classes
   */
  className: PropTypes.string,
  /**
   * Optional data-name attribute
   * (use for testing/selection)
   */
  dataName: PropTypes.string,
};

Header.defaultProps = {
  children: null,
  className: null,
  dataName: 'card-header',
};

export default Header;
