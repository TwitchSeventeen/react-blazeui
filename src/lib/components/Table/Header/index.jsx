import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function Header(props) {
  const {
    children, className, ...other
  } = props;

  const classes = classNames(
    'c-table__head',
    className,
  );

  return (
    <thead className={classes} {...other}>
      { children }
    </thead>
  );
}

Header.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
  /**
   * Additional classes
   */
  className: PropTypes.string,
};

Header.defaultProps = {
  className: null,
  children: null,
};

export default Header;
