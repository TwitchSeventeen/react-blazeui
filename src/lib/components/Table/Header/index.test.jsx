import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Header from './index';

describe('Header', () => {
  it('should be defined', () => {
    expect(Header).toBeDefined();
  });

  it('renders correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
      <Header>
        lorem ipsum dolor
      </Header>,
    );

    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
