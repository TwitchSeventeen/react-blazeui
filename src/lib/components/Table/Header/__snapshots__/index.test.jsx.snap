// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<thead
  className="c-table__head"
>
  lorem ipsum dolor
</thead>
`;
